controlplane/src/core/repositories/analytics/TraceRepository.ts

```diff
@@ -78,4 +78,45 @@ export class TraceRepository {
       },
     }));
   }
+
+  public async getRouterVersion(organizationID: string, startDate: string): Promise<string> {
+    const parsedDateRange = isoDateRangeToTimestamps({ start: startDate, end: startDate });
+    const [start] = getDateRange(parsedDateRange);
+
+    const queryWithTimeConstraint = `
+    WITH 
+      toDateTime('${start}') AS startDate
+    SELECT  
+        SpanAttributes['wg.router.version'] as routerVersion
+    FROM ${this.client.database}.otel_traces
+    WHERE Timestamp >= startDate AND SpanAttributes['wg.router.version'] != '' AND SpanAttributes['wg.organization.id'] = '${organizationID}' 
+    ORDER BY Timestamp ASC
+    LIMIT 1 OFFSET 0
+    `;
+
+    const res = await this.client.queryPromise(queryWithTimeConstraint);
+
+    if (Array.isArray(res)) {
+      if (res.length === 0) {
+        const query = `
```

Comment:
What's the reason behind this? If a router version can't be found in the range of the UI we will look for any version?

Reply:
as we are checking the otel traces table, if the users have not run an operation after the start date, I assume that whatever version they were using before the start date as the router version

Reply:
The problem is that this will result in a fullscan. If no router version can be found after the composition, I'd communicate it "No activity"